runtime/runtime-params-estimator/src/gas_metering.rs

```diff
@@ -237,9 +238,16 @@ pub fn compute_gas_metering_cost(
         );
         assert!(result.1.is_none());
     }
-    let total_raw_no_gas = start.elapsed().scalar_cost().to_i128().unwrap();
+    let total_raw_no_gas = start.elapsed().to_gas();
 
-    // println!("with gas: {}; no gas {}", total_raw_with_gas, total_raw_no_gas);
+    // TODO: This seems to fail almost always but has some non-determinism to it.
+    assert!(
```

Comment:
This is what I am looking at right now with higher priority than the general refactoring, as it seems related to RocksDB compaction. Here. it looks like the IO costs are sometimes attributed to one and sometimes to the other execution. It's always 64kiB for one and 0B for the other. The order in which the calls with/without gas metering happen does not affect it that much. If at least one of the test iterations has the IO cost attributed to the case without gas-metering, then this check fails.

This test also fails in the CI.

Reply:
Uhu, let's than maybe comment this out and land just the refactor?

Reply:
Only commenting the assertion out won't help, the subtraction below will panic with an underflow. But I can add `#[ignore]` to the affected tests

Reply:
Ah never mind, those tests can be removed anyway